minorminer/utils/zephyr/node_edge.py

```diff
@@ -0,0 +1,635 @@
+                    f"shape and ccoord must be both quotient or non-quotient, got {self._shape}, {new_ccoord}"
+                )
+            if (self._shape.t is not None) and (new_ccoord.k not in range(self._shape.t)):
+                raise ValueError(f"Expected k to be in {range(self._shape.t)}, got {new_ccoord.k}")
+
+            # check x, y value of CartesianCoord is consistent with m
+            if self._shape.m is not None:
+                if all(val in range(4 * self._shape.m + 1) for val in (new_ccoord.x, new_ccoord.y)):
+                    self._ccoord = new_ccoord
+                else:
+                    raise ValueError(
+                        f"Expected ccoord.x and ccoord.y to be in {range(4*self._shape.m+1)}, got {new_ccoord.x, new_ccoord.y}"
+                    )
+        self._ccoord = new_ccoord
+
+    @staticmethod
+    def get_coord(coord: tuple[int]) -> CartesianCoord | ZephyrCoord:
+        """Takes a tuple[int] and returns the corresponding CartesianCoord
+        or ZephyrCoord"""
+        if (not isinstance(coord, tuple)) or (not all(isinstance(c, int) for c in coord)):
+            raise TypeError(f"Expected {coord} to be a tuple[int], got {coord}")
+        if any(c < 0 for c in coord):
+            raise ValueError(f"Expected elements of coord to be non-negative, got {coord}")
+        len_coord = len(coord)
+        if len_coord in (2, 3):
+            x, y, *k = coord
+            if x % 2 == y % 2:
+                raise ValueError(f"Expected x, y to differ in parity, got {x, y}")
+            return CartesianCoord(x=x, y=y) if len(k) == 0 else CartesianCoord(x=x, y=y, k=k[0])
+        if len_coord in (4, 5):
+            u, w, *k, j, z = coord
+            for var, val in {"u": u, "j": j}.items():
+                if not val in [0, 1]:
+                    raise ValueError(f"Expected {var} to be in [0, 1], got {val}")
+            return (
+                ZephyrCoord(u=u, w=w, j=j, z=z)
+                if len(k) == 0
+                else ZephyrCoord(u=u, w=w, k=k[0], j=j, z=z)
+            )
+        raise ValueError(f"coord can have length 2, 3, 4 or 5, got {len_coord}")
+
+    @property
+    def zcoord(self) -> ZephyrCoord:
+        """Returns ZephyrCoordinate corresponding to ccoord"""
+        return cartesian_to_zephyr(self._ccoord)
+
+    def is_quo(self) -> bool:
+        """Decides if the ZNode object is quotient"""
+        return (self._ccoord.k is None) and (self._shape.t is None)
+
+    def to_quo(self) -> ZNode:
+        """Returns the quotient ZNode corresponding to self"""
+        qshape = ZShape(m=self._shape.m)
+        qccoord = CartesianCoord(x=self._ccoord, y=self._ccoord)
+        return ZNode(coord=qccoord, shape=qshape, convert_to_z=self.convert_to_z)
+
+    @property
+    def direction(self) -> int:
+        """Returns direction, 0 or 1"""
+        return self._ccoord.x % 2
+
+    def is_vertical(self) -> bool:
+        """Decides whether self is a vertical qubit"""
```

Comment:
`self` is a node, right?
```suggestion
        """Decides whether node is a vertical qubit"""
```
Similarly for docstrings below.

Reply:
Done in https://github.com/dwavesystems/minorminer/pull/261/commits/70d84ed616e04331a0f181015fdf1fb3e2774596